lucet-wasi/src/c_api.rs

```diff
@@ -27,15 +27,18 @@ pub unsafe extern "C" fn lucet_wasi_ctx_args(
     assert_nonnull!(wasi_ctx);
     let mut b = Box::from_raw(wasi_ctx as *mut WasiCtxBuilder);
     let args_raw = std::slice::from_raw_parts(argv, argc);
-    let args: Result<Vec<&str>, _> = args_raw
+    let args: Result<Vec<String>, std::str::Utf8Error> = args_raw
         .iter()
-        .map(|arg| CStr::from_ptr(*arg).to_str())
+        .map(|arg| Ok(CStr::from_ptr(*arg).to_str()?.to_owned()))
```

Comment:
This seems fine, I'm just curious what motivated the switch?

Reply:
Honestly, I just forgot to make the args and env builders take an `AsRef<str>` and didn't want to make another upstream PR just to get this in. There wasn't any good reason for it.